hist/histv7/inc/ROOT/RAxis.hxx

```diff
@@ -96,6 +96,67 @@ protected:
       return (int)rawbin;
    }
 
+   /// Check if two axis have the same bin borders
+   ///
+   /// Default implementation should work for any RAxis type, but is a little
+   /// bit stupid. RAxis implementations are encouraged to provide optimized
+   /// overrides for common comparison scenarios.
+   virtual bool HasSameBinBordersAs(const RAxisBase& other) const noexcept {
+      // Axis growability (and thus under/overflow bin existence) must match
+      if (CanGrow() != other.CanGrow()) return false;
+
+      // Number of normal bins must match
+      if (GetNBinsNoOver() != other.GetNBinsNoOver()) return false;
+
+      // Left borders of normal bins must match
+      for (int bin: *this) {
+         if (GetBinFrom(bin) != other.GetBinFrom(bin)) return false;
+      }
+
+      // Right border of the last normal bin (aka maximum) must also match
+      if (GetMaximum() != other.GetMaximum()) return false;
+
+      // If all of these checks passed, the two axes have the same bin borders
+      return true;
+   }
+
+   /// If supplementary bin metadata (e.g. bin labels) exists in `this`, make
+   /// sure that `other` provides the same bin metadata.
+   ///
+   /// The implementation can work under the assumption that the number of bins
+   /// and bin borders of `this` and `other` has already been compared equal.
+   ///
+   /// RAxis classes which override a non-default implementation of
+   /// `HasSameBinMetadataAs` should start by calling the parent class'
+   /// implementation of this method, in order to allow "stacking" of bin
+   /// metadata across the class hierarchy.
+   ///
+   /// Adding an override of `HasSameBinMetadataAs` to an RAxis class is
+   /// considered an API-breaking change: subclasses can assume that if their
+   /// parent classes don't override the default implementation of
+   /// `HasSameBinMetadataAs`, they never will.
```

Comment:
The following properties are mutually exclusive:

1. Provide a default implementation of `AlsoHasSameBinMetadataAs` that defers to `HasSameBinMetadataAs`, so that users only need to override `AlsoHasSameBinMetadataAs` in addition to `HasSameBinMetadataAs` if there is a performance benefit in doing so.
2. Unconditionally advise subclasses to call the superclass' implementation of `AlsoHasSameBinMetadataAs` in their `AlsoHasSameBinMetadataAs` overrides, without losing all performance benefits from the `AlsoHasSameBinMetadataAs` optimization.

Technically, `HasSameBinMetadataAs` is not subjected to this problem, but I thought it was better to apply the same rule to both methods from an ergonomics point of view.

So far, I prioritized property 1, but I could be convinced that property 2 is more important and we should pick that one.

Reply:
Will be resolved by the removal of the metadata mechanism, as mentioned in the other thread.

Reply:
De facto resolved by d3ac8bf85bdbe .